bin/pre-commit-check.sh

```diff
@@ -0,0 +1,126 @@
+#!/bin/bash
+set -euo pipefail
+
+printf "\n"
+
+# Function to extract the network ID from a broadcast file path
+get_network_id() {
+    local file_path="$1"
+    local network_id=$(echo "$file_path" | sed -E 's/.*\/([0-9]+)\/.*/\1/')
+    if [ -z "$network_id" ]; then
+        echo "Error: Could not extract network ID from file path: $file_path" >&2
+        return 1
+    fi
+    echo "$network_id"
+}
+
+# Function to map network ID to network name
+get_network_name() {
+    local network_id="$1"
+    case "$network_id" in
+        "42220")
+            echo "celo"
+            ;;
+        "44787")
+            echo "alfajores"
+            ;;
+        "62320")
+            echo "baklava"
+            ;;
+        *)
+            echo "Error: Unsupported network ID: $network_id" >&2
+            exit 1
+            ;;
+    esac
+}
+
+# Fetch addresses of newly deployed contracts from broadcast file
+fetch_addresses() {
+    cat "${broadcast_file}" |
+    jq -r '.transactions[] | select(.transactionType == "CREATE") | .contractAddress' |
+    grep -v null
+}
```

Comment:
There's also factory contract case, where calling a function on a contract results in contract creation. 
Foundry verifies those contracts on Sourcify when executing the transaction, and our Sourcify->CeloScan script also picks them up. You can look at the `broadcasts/ExecuteProposal.sol/42220/MUGOV.json` to see the structure. Should be easy to concatenate.

Reply:
good catch, i extended the jq expression to include factory deployments as well